src/pages/index.js

```diff
@@ -0,0 +1,627 @@
+      confirmCardDeleteButton.textContent = "Delete";
+      // TODO: Close the modal.
+      closeModal(modalNewPost);
+    });
+}
+//Delete card function
+function deleteCard(card) {
+  card.remove(card);
+}
+
+confirmCardDeleteButton.addEventListener("click", handleDeleteSubmit);
+cancelCardDeleteButton.addEventListener("click", () =>
+  closeModal(modalDeleteCard)
+);
+//Desplay inital cards or added card
+function displayCard(data) {
+  if (Array.isArray(data)) {
+    data.forEach((card) => {
+      // Pass the array item to your getCardElement() function to create a card element.
+      const cardDisplay = getCardElement(card);
+      // Use the appropriate built-in DOM method to add this HTML element to the page.
+      cardsElement.append(cardDisplay);
+    });
+
+    // This is for user input single card, separate from initial cards' array
+  } else {
+    const cardDisplay = getCardElement(data);
+    cardsElement.prepend(cardDisplay);
+  }
+}
+
+// Call display card funtion to display initial cards' array
+// displayCard(initialCards);
+
+//Make card from template function
+function getCardElement(cardData) {
+  const cardElement = cardsElement
+    .querySelector("#card")
+    .content.querySelector(".card")
+    .cloneNode(true);
+  // select the card title and image and store them in variables
+  const cardImage = cardElement.querySelector(".card__image");
+  cardImage.src = cardData.link;
+  cardImage.alt = cardData.name;
+  cardElement.querySelector(".card__description").textContent = cardData.name;
+
+  //Set love button
+  const loveButton = cardElement.querySelector(".card__love-button");
+
+  // Check if current user has already liked this card
+  if (cardData.isLiked) {
+    loveButton.classList.add("card__love-button_activated");
+  }
+  loveButton.addEventListener("click", () => {
+    activateLoveButton(loveButton, cardData._id);
+  });
+  // Set delete button
+  const deleteButton = cardElement.querySelector(".card__delete-button");
+
+  // Modify it so that you can pass the card element as an argument, as well
+  // as the card data.
+  deleteButton.addEventListener("click", () => {
+    selectedCard = cardElement; // Assign the card element to selectedCard
+    selectedCardId = cardData._id; // Assign the card's ID to selectedCardId
+
+    // open the delete confirmation modal
+    openModal(modalDeleteCard);
+  });
+  // Set image click button
+  // const imageButton = cardElement.querySelector(".card__image-button");
+  // imageButton.addEventListener("click", () => {
+  //   clickImage(cardElement, cardImage);
+  // }); Instead of creeating a button just listen on the image element click
+  cardImage.addEventListener("click", () => {
+    clickImage(cardElement, cardImage);
+  });
+
+  //return the ready HTML element with the filled-in data
+  return cardElement;
+}
+
+//Activate love button funtion
+function activateLoveButton(loveButton, cardId) {
+  loveButton.classList.toggle("card__love-button_activated");
```

Comment:
You need to **toggle likes** only in a `then` block after a successful response from the server because if there is an error, the user will not know about that and will see the toggled heart color.